tf_quant_finance/black_scholes/vanilla_prices.py

```diff
@@ -97,21 +117,35 @@ def option_price(volatilities,
         volatilities, dtype=dtype, name='volatilities')
     expiries = tf.convert_to_tensor(expiries, dtype=dtype, name='expiries')
 
-    if discount_factors is None:
-      discount_factors = tf.convert_to_tensor(
-          1.0, dtype=dtype, name='discount_factors')
+    if risk_free_rates is not None:
+      risk_free_rates = tf.convert_to_tensor(
+          risk_free_rates, dtype=dtype, name='risk_free_rates')
+    elif discount_factors is not None:
```

Comment:
you might as well raise an error if both discount_factors and risk_free_rates are supplied. This guards from inconsistent values of discount_factors and risk_free_rates

Reply:
I do a few lines above!